0019-refactor-image-operations/README.md

```diff
@@ -0,0 +1,315 @@
+is ready for each stage.
+
+Consider the following stages when setting graduation criteria:
+- Alpha: Feature is behind a feature flag, basic tests in place.
+- Beta: Gather feedback from users, complete core features, add more tests.
+- GA: Prove real-world usage, complete rigorous testing, gather feedback.
+
+In general, features should wait at least two releases between Beta and GA to
+allow time for feedback. For features moving to GA, include conformance tests
+to ensure stability and compatibility.
+
+#### Deprecation
+If this feature will eventually be deprecated, plan for it:
+- Announce deprecation and support policy.
+- Wait at least two versions before fully removing it.
+-->
+
+As this is not a feature, but a refactor of an implementation this change will be introduced as stable immediately. 
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+proposal:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  package definition or deployment required to make on upgrade, in order to
+  maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  package definition or deployment required to make on upgrade, in order to
+  make use of the proposal?
+-->
+
+N/A
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+proposal:
+- Does this proposal involve coordinating behavior between components?
+  - (i.e. the Zarf Agent and CLI? The init package and the CLI?)
+-->
+
+There are differences between the annotations on the index.json for images pulled by Crane vs ORAS. Zarf must ensure that packages created using Crane are backwards compatible for all operations.
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a ZEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling acceptance of the ZEP
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Zarf release where an initial version of the ZEP was available
+- the version of Zarf where the ZEP graduated to general availability
+- when the ZEP was retired or superseded
+-->
+
+- 2025-03-05: Document created
+
+## Drawbacks
+
+<!--
+Why should this ZEP _not_ be implemented?
+-->
+
+There is potential for image operations to be slower for some use cases since we are defaulting to less concurrency. This is a lower risk than the current reliability issues that users face which can completely break create + deploy operations. 
```

Comment:
Potentially a future concern, but is it feasible to analyze the pkg layer count and size before pushing or pulling, and adapt our concurrency right away? E.g. pull many small layers at a time, then prioritize large images if there's one or more in a package?

Reply:
I do think it would be possible, but IMO the added complexity isn't worth it, at least yet. It gets tricky because in order for a real gain to be seen by doing this, we would have to start concurrently pulling images and image layers. Even though ORAS does handle pulling multiple images, it takes some work on our part and makes the code considerably more complex.

I will mention this in the alternatives.